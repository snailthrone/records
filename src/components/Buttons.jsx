import React from 'react';
import styled from 'styled-components';

import Input from './Input';

const Buttons = styled.div`
  background: inherit;
  bottom: 0;
  left: 0;
  margin: auto;
  position: fixed;
  width: 100%;
  z-index: 100;
`;

const Button = styled.div`
  background: #222;
  color: #fff;
  display: inline-block;
  font-weight: bold;
  margin: auto;
  padding: 20px 0;
  text-align: center;
  top: 20px;
  width: ${100/3}%;
  &:hover {
    background: #7F7E7E;
    cursor: pointer;
  }
`;

const Label = styled.label``;

export default props => (
  <Buttons>
    <Input onChange={props.searchContent} value={props.value}/>
    <Button onClick={event => props.changeList('collection')}>Collection</Button>
    <Button onClick={event => props.changeList('upcoming')}>Upcoming</Button>
    <Button onClick={event => props.changeList('wantlist')}>Wishlist</Button>
  </Buttons>
);