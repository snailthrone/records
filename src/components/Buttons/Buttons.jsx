import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'

import Input from '../Input/Input'

const Wrapper = styled.div`
  background-color: inherit;
  bottom: 0;
  left: 0;
  margin: auto;
  position: fixed;
  width: 100%;
  z-index: 100;
`

const Button = styled.div`
  background-color: #222222;
  color: #ffffff;
  display: inline-block;
  font-family: 'Questrial', Verdana, sans-serif;
  font-size: 14px;
  font-weight: bold;
  margin: auto;
  padding: 20px 0;
  text-align: center;
  width: ${100 / 4}%;
  &:hover {
    background: #7F7E7E;
    cursor: pointer;
  }
`

const Buttons = ({ changeList, searchContent, value }) => (
  <Wrapper>
    <Input
      className="app__search-field"
      onChange={searchContent}
      value={value}
    />
    <Button
      className="app__button"
      onClick={() => changeList('collection')}
    >
      CD
    </Button>
    <Button
      className="app__button"
      onClick={() => changeList('upcoming')}
    >
      Coming
    </Button>
    <Button
      className="app__button"
      onClick={() => changeList('wishlist')}
    >
      Wishlist
    </Button>
    <Button
      className="app__button"
      onClick={() => changeList('vinyls')}
    >
      Vinyls
    </Button>
  </Wrapper>
)

Buttons.propTypes = {
  changeList: PropTypes.func.isRequired,
  searchContent: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired,
}

export default Buttons
