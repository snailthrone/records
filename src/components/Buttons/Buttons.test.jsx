import React from 'react'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'

import Buttons from './Buttons'

describe('<Buttons />', () => {
  const changeList = jest.fn()

  it('Renders correctly', () => {
    const tree = renderer.create(
      <Buttons
        changeList={changeList}
        searchContent={() => {}}
        value="Opeth"
      />,
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Handles the click events', () => {
    const wrapper = mount(
      <Buttons
        changeList={changeList}
        searchContent={() => {}}
        value="Opeth"
      />,
    )

    const buttons = wrapper.find('.app__button')
    buttons.forEach((button) => {
      button.simulate('click')
      expect(changeList).toHaveBeenCalledWith('collection')
    })
  })
})
