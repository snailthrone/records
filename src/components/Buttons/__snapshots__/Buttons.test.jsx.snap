// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Buttons /> Renders correctly 1`] = `
<div
  className="Wrapper-t8at1t-0 cEVezx"
>
  <div
    className="Container-s1cvj67l-0 kUgPaT"
  >
    <input
      aria-label="search"
      className="Input-s1cvj67l-1 eaTpaa"
      name="search-field"
      onChange={[Function]}
      placeholder="Search"
      type="text"
      value="Opeth"
    />
  </div>
  <div
    className="app__button Button-t8at1t-1 fASBcw"
    onClick={[Function]}
  >
    CD
  </div>
  <div
    className="app__button Button-t8at1t-1 fASBcw"
    onClick={[Function]}
  >
    Coming
  </div>
  <div
    className="app__button Button-t8at1t-1 fASBcw"
    onClick={[Function]}
  >
    Wishlist
  </div>
  <div
    className="app__button Button-t8at1t-1 fASBcw"
    onClick={[Function]}
  >
    Vinyls
  </div>
</div>
`;
